agent/src/main/java/org/openjdk/jmc/agent/converters/FileConverter.java

```diff
@@ -32,16 +33,23 @@
  */
 package org.openjdk.jmc.agent.converters;
 
+import java.io.File;
+
+import org.openjdk.jmc.agent.Agent;
+
 /**
- * Interface for converting an object to a double.
+ * A converter which converts files into their canonical path.
  */
-public interface DoubleConverter<T> {
-	/**
-	 * Converts an object to a double.
-	 * 
-	 * @param o
-	 *            the object to convert.
-	 * @return the object converted to a double.
-	 */
-	double convert(T o);
+public final class FileConverter {
+	public static String convert(File file) {
+		if (file == null) {
+			return "null";
+		}
+		try {
+			return file.getCanonicalPath();
+		} catch (Throwable e) {
+			Agent.getLogger().warning("Agent failed to convert file to String: " + e.getMessage());
```

Comment:
Should propagate the original exception instead of just its message?

Reply:
I thought it might be annoying, since you know where you've declared them. But perhaps I should - there might be multiple locations. Another variant would be to silently ignore and just return the path, but perhaps you do want to know.

Reply:
Fixed.